### PR TITLE
[INLONG-3274][Agent]  Change the task status to failed when the Kafka topic was deleted

### LINKED ISSUE #3274: [Bug][Agent] When Kafka topic is deleted, then change task status to failed.
### What happened

when kafka topic deleted,then chang task status to failed

### What you expected to happen

when kafka topic deleted,then chang task status to failed

### How to reproduce

when kafka topic deleted,then chang task status to failed

### Environment

_No response_

### InLong version

0.12.0

### InLong Component

InLong Agent

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zk1510**
### Title Name: [INLONG-3274][Agent] Fix change task status failed when the Kafka topic was deleted

Fixes #3274

### Motivation

When Kafka topic is deleted, then change task status to failed.

### Modifications

When Kafka topic is deleted, then change task status to failed.

### CODE REVIEW

**Review by healchow — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Comment by dockerzhang** [comment]
@EMsnap PTAL

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/KafkaReader.java [hunk 1]
```diff
@@ -143,7 +148,9 @@ public Message read() {
             }
         } else {
             // commit offset
-            consumer.commitAsync();
+            if (!isSourceExist()) {
```
Comment (EMsnap) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/KafkaReader.java:151]: when source not exist commit offset? should commit offset when source exist

**Review by EMsnap — COMMENTED** [review]
PTAL

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/KafkaReader.java (hunk 1 above)
Comment (zk1510) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/KafkaReader.java:151]: you shou see the method isSourceExist()

Comment (zk1510) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/KafkaReader.java:151]: isSourceExist() return false indicates that  source does not exist

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/plugin/Reader.java [hunk 2]
```diff
@@ -63,4 +63,10 @@
      * finish read
      */
     void finishRead();
+
+    /**
+     * source deleted
```
Comment (EMsnap) [inline inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/plugin/Reader.java:68]: source is exist

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/KafkaSource.java [hunk 3]
```diff
@@ -107,9 +105,15 @@ public KafkaSource() {
             // example:0#110_1#666_2#222
             partitionOffsets = allPartitionOffsets.split(JOB_OFFSET_DELIMITER);
         }
+        //
+        props.put(KAFKA_SESSION_TIMEOUT, 30000);
```
Comment (EMsnap) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/KafkaSource.java:109]: what's the purpose of doing this?

**Review by EMsnap — COMMENTED** [review]
PLAL

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/KafkaSource.java (hunk 3 above)
Comment (zk1510) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/KafkaSource.java:109]: set the kafka consumer session timeout

**Review by EMsnap — APPROVED** [review]

**Pull request merged**